README.md

```diff
@@ -19,8 +19,240 @@
+Once the DAG has been completed (and with no errors), Immucore its finished, and it's ready for the initramfs init process to do a switch_root and pivot into the final root to boot the system.
+
+When booting from Netboot/LiveCD/Do nothing (`rd.cos.disable` or `rd.immucore.disable` on the cmdline) the DAG It's pretty simple. 
+It proceeds to create a sentinel file under `/run/cos/` with the boot mode (`live_mode`) so cloud configs can identify that they are booting from live media and ends.
+
+
+When booting from active/passive/recovery the DAG gets a bit more complicated. You can see the default DAG for an active/passive/recovery system by running immucore with `--dry-run`
+
+```bash
+1.
+ <init> (background: false) (weak: false)
+2.
+ <mount-state> (background: false) (weak: false)
+ <mount-base-overlay> (background: false) (weak: false)
+ <mount-tmpfs> (background: false) (weak: false)
+ <create-sentinel> (background: false) (weak: false)
+3.
+ <discover-state> (background: false) (weak: false)
+4.
+ <mount-root> (background: false) (weak: false)
+5.
+ <mount-oem> (background: false) (weak: false)
+6.
+ <rootfs-hook> (background: false) (weak: false)
+7.
+ <load-config> (background: false) (weak: false)
+8.
+ <custom-mount> (background: false) (weak: false)
+ <overlay-mount> (background: false) (weak: false)
+9.
+ <mount-bind> (background: false) (weak: false)
+10.
+ <write-fstab> (background: false) (weak: true)
+```
+
+As shown in the DAG, the steps are in order and that shows their dependencies, i.e. `mount-root` depends on `discover-state`and that is why it's just below it.
+It won't run until the previous step has completed **without errors**.
+There is also the `weak` value which indicates that this step has weak dependencies. It will run even if its dependencies failed, instead of refusing to run.
+
+
+
+### Steps explained
+
+ - `mount-state`: Will mount the `COS_STATE` partition under `/run/initramfs/cos-state`
+ - `mount-tmpfs`: Will mount `/tmp` 
+ - `create-sentinel`: Will create the sentinel file identifying the boot mode (`active_mode`, `passive_mode`, `recovery_mode` or `live_mode`) under `/run/cos/`
+ - `mount-base-overlay`: Will mount the base overlay under `/run/overlay`
+ - `discover-state`: Will find the correct image under `/run/initramfs/cos-state` and mount it as a loop device
+ - `mount-root`: Will mount the `/dev/disk/by-label/$LABEL` device under the sysroot (Usually `/sysroot`). This label is set in grub depending on the selected entry, as part of the cmdline (i.e. `root=LABEL=COS_ACTIVE`) 
+ - `mount-oem`: Will **try** to mount the oem label device under `/sysroot/oem`. This label is set in grub by default (`rd.cos.oemlabel=COS_OEM`) but also on the default `cos-layout.env` file with Kairos. This partition is not mandatory so It's allowed to fail.
```

Comment:
all others are not using dots, feel free to either remove this one or add to the others

```suggestion
 - `mount-oem`: Will **try** to mount the oem label device under `/sysroot/oem`. This label is set in grub by default (`rd.cos.oemlabel=COS_OEM`) but also on the default `cos-layout.env` file with Kairos. This partition is not mandatory so It's allowed to fail
```